### PR TITLE
Adding versioning to S3 origin bucket

### PR DESCRIPTION
**Pull request by hawkesnc**
## what
* This change allows you to turn on versioning for objects in the bucket

## why
* For building static websites, we require the use of s3+cdn that this module provides. However, in some instances, we would like to ability to roll back to a previous version. 
* Since this is meant for a "dev" like environment, we set the cdn ttl to very low to pull objects back from the s3 bucket.
* The reason we use the CDN is to support TLS since s3 does not.

## references
* N/A

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -123,6 +123,14 @@ resource "aws_s3_bucket" "origin" {
     }
   }
 
+  dynamic "versioning" {
+    for_each = var.versioning_enabled ? ["true"] : []
```
Comment (nitrocode) [inline main.tf:127]: This can be simplified to this
```suggestion
    for_each = [var.versioning_enabled]
```

File: main.tf [hunk 2]
```diff
@@ -123,6 +123,14 @@ resource "aws_s3_bucket" "origin" {
...
+
+    content {
+      enabled = var.versioning_enabled
```
Comment (nitrocode) [inline main.tf:130]:
```suggestion
      enabled = versioning.value
```

**Comment by hawkesnc** [comment]
Added those changes in. Thanks @nitrocode for the review! 👍

**Comment by hawkesnc** [comment]
Is this good to merge?

**Comment by hawkesnc** [comment]
Sweet! Thanks 👍

**Comment by Gowiem** [comment]
@hawkesnc Thanks for the contribution! Released as [0.34.0](https://github.com/cloudposse/terraform-aws-cloudfront-s3-cdn/releases/tag/0.34.0).

**Pull request merged**